backend/src/run.py

```diff
@@ -294,6 +294,7 @@ async def run_individual(request: Request):
         return json({"success": True, "data": None})
     except Exception as exception:
         logger.error(exception, exc_info=True)
+        del ctx.cache[full_data["id"]]
```

Comment:
Instead of removing it *after* something went wrong, it's a lot more robust to remove the cached value *before* running that node. Actually, even before validating inputs.

Reply:
Whats the point of catching it if we're just going to remove it before running it?